### DISCUSSION TITLE
Previous command ID, red if return code was 0

### DISCUSSION THREAD

**Discussion by flickerfly** [starship/starship#4440] [Q&A] [answered]
I like to include in my prompt the previous command's history ID to make it easier to refer back to with a commands like `!34` and color that according to its success or failure based on return code. Is this something I'd need to write something custom for?

**Comment by chipbuster** [comment, marked as answer]
Yes, you'll have to make a custom module for that. starship does not have ability to access the history ID of shells (in fact, this was one of the rationales of allowing custom modules to unescape their contents).

**Reply by flickerfly** [reply]
Thanks
